src/sugar3/activity/widgets.py

```diff
@@ -396,3 +396,31 @@ def __init__(self):
         self.paste = PasteButton()
         self.insert(self.paste, -1)
         self.paste.show()
+
+class CollabTextEditor(Gtk.Box):
+    def __init__(self,activity):
+        Gtk.Box.__init__(self)
+        self._collab = CollabWrapper(activity)
+        self._collab.connect('message', self.__message_cb)
+        self._collab.setup()
+        self.textview = Gtk.TextView()
+        self.textview.set_editable(True)
+        self.textview.set_cursor_visible(True)
+        self.textview.set_wrap_mode(Gtk.WrapMode.WORD)
+        self.textbuffer = self.textview.get_buffer()
+        self.textbuffer.connect('changed', self.__text_buffer_edited_cb)
+        self.textbuffer.set_text("TESTING")
```

Comment:
Please make it empty by default.  Also, please deal with if a new buddy joins - you need to transfer the text to them.